src/main/scala/com/twitter/finagle/postgres/PostgresClientImpl.scala

```diff
@@ -152,25 +154,32 @@ class PostgresClientImpl(
    * Issue a single, prepared SELECT query and wrap the response rows with the provided function.
    */
   override def prepareAndQuery[T](sql: String, params: Param[_]*)(f: Row => T): Future[Seq[T]] = {
-    for {
-      service <- factory()
-      statement = new PreparedStatementImpl("", sql, service)
-      result  <- statement.select(params: _*)(f)
-    } yield result
+    typeMap().flatMap { _ =>
```

Comment:
Not sure these `flatMap`s are strictly necessary, they'll cause it to fail slightly sooner but it should fail anyway when `typeMap()` is accessed for real later down in the code.

Reply:
I rather keep this, I know its not super necessary but I rather have it failing sooner.